### PR TITLE
Add Jenkinsfile and Dockerfile to build the Modbus simulator

### LINKED ISSUE #135: Add Jenkinsfile and Dockerfile to build the Modbus simulator
Add Jenkinsfile and Dockerfile to build the Modbus simulator.
Example: https://github.com/edgexfoundry/ci-build-images/tree/edgex-compose

### PR DESCRIPTION
**Pull request by weichou1229**
Add Jenkinsfile and Dockerfile to build the Modbus simulator.

Fix #135

### CODE REVIEW

**Comment by weichou1229** [comment]
@jamesrgregg 
> My suggestion is to build with the modbus image. That way you keep both code bases together. You can work with @Iain Anderson to creat the branch within that mod us repo. It doesn’t belong with ci-build images.

As your suggestion, we create a branch and PR to build the image but got the Jenkins job error.
Could you please help resolve the issue?

**Comment by cloudxxx8** [comment]
The error is caused by snap build, and it won't prevent the code merge.

File: Dockerfile [hunk 1]
```diff
@@ -14,29 +14,28 @@
...
 
-ARG MAKE='make build'
+WORKDIR /go/src/github.com/edgexfoundry/device-modbus-go/simulator
 
 RUN sed -e 's/dl-cdn[.]alpinelinux.org/nl.alpinelinux.org/g' -i~ /etc/apk/repositories
-RUN apk add --update --no-cache make git openssh build-base
 
-# set the working directory
-WORKDIR $GOPATH/src/github.com/edgexfoundry/device-modbus-go
+RUN apk update && apk add zeromq-dev libsodium-dev pkgconfig build-base git
```
Comment (cloudxxx8) [inline Dockerfile:23]: do you need zeromq-dev here?

Comment (weichou1229) [reply Dockerfile]: I don't need the zeromq-dev, removed

**Comment by jamesrgregg** [comment]
Thanks for moving this automation to this repo. @cloudxxx8 @weichou1229 
I see the Pipeline build was Successful

**Pull request merged**